src/components/molecules/Common/AssetModal/VirtualAssetCard/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback, useState, useEffect } from "react";
+import { styled } from "@reearth/theme";
+
+type CardSize = "small" | "medium" | "large" | "list";
+
+export type Props = {
+  assetsLength?: number;
+  containerWidth?: number;
+  cardSize?: CardSize;
+};
+
+const VirtualAssetCard: React.FC<Props> = ({ assetsLength, containerWidth, cardSize }) => {
+  const [virtualWidth, setVirtualWidth] = useState(0);
+
+  const handleVirtualWidth = useCallback(() => {
+    if (containerWidth) {
+      if (assetsLength) {
+        const cardWidth = cardSize === "medium" ? 216 : 128;
+        const numInrow = Math.floor(containerWidth / cardWidth);
+        const lastRowNum = assetsLength % numInrow;
+        const gapWidth = (containerWidth - cardWidth * numInrow) / (numInrow - 1);
+        const virtualBoxNumber = numInrow - lastRowNum;
+        const virtualBoxWidth = virtualBoxNumber * cardWidth + gapWidth * (virtualBoxNumber - 1);
+        if (lastRowNum === 0) {
+          setVirtualWidth(0);
+        } else {
+          setVirtualWidth(virtualBoxWidth);
+        }
+      }
+    }
+  }, [assetsLength, cardSize, containerWidth]);
+
+  useEffect(() => {
+    handleVirtualWidth();
```

Comment:
You don't need to use useCallback, you can just write the contents of the function here.